src/features/auth/authSlice.js

```diff
@@ -224,6 +228,53 @@ export const removeBookmark = createAsyncThunk(
   }
 );
 
+//follow a user
+export const followUser = createAsyncThunk(
+  "auth/followUser",
+  async (followuserId, { getState }) => {
+    const userstate = getState();
+    const currentUserId = userstate.auth.user.userId;
+    try {
+      const userRef = doc(database, "users", currentUserId);
+       await updateDoc(userRef, {
+        following: arrayUnion(followuserId),
+      });
+
+      const followrUserRef = doc(database, "users", followuserId);
+      await updateDoc(followrUserRef, {
+        followers: arrayUnion(currentUserId),
+      });
+      return { followuserId, userId: currentUserId };
+    } catch (error) {
+      console.error(error);
+      return Promise.reject(error);
+    }
+  }
+);
+
+// unfollow a user
+export const unfollowUser = createAsyncThunk(
+  "auth/unfollowUser",
+  async (followuserId, { getState }) => {
+    const userstate = getState();
+    const currentUserId = userstate.auth.user.userId;
+    try {
+      const userDataRef = doc(database, "users", currentUserId);
+      await updateDoc(userDataRef, {
+        following: arrayRemove(followuserId),
+      });
+      const followerUserRef = doc(database, "users", followuserId);
+       await updateDoc(followerUserRef, {
+        followers: arrayRemove(currentUserId),
+      });
+      return { followuserId, userId: currentUserId };
+    } catch (error) {
+      console.error(error);
+      return Promise.reject(error);
+    }
+  }
+);
+
```

Comment:
Had a query on when to make new slices in React redux state management format?
Because you have kept the feature of follow/unfollow in the auth Slice itself instead of making a new slice.
**Edit**: I think it is because in this auth slice you are creating a user itself from this slice.